include/seqan3/range/container/small_string.hpp

```diff
@@ -0,0 +1,328 @@
+     *
+     * Strong exception guarantee. No data is modified.
+     *
+     * ### Complexity
+     *
+     * Linear in the size of the string.
+     */
+    std::string str() const
+    {
+        return std::string{this->cbegin(), this->cend()};
+    }
+
+    /*!\brief Returns the content represented as 0-terminated c-style string.
+     *
+     * \returns `char const *` The stored string.
+     *
+     * ### Exceptions
+     *
+     * No-throw guarantee.
+     *
+     * ### Complexity
+     *
+     * Constant.
+     */
+    constexpr char const * c_str() const noexcept
+    {
+        return data_.data();
+    }
+    //!\}
+
+    /*!\name Modifiers
+     * \{
+     */
+    //!\copydoc seqan3::small_vector::clear
+    constexpr void clear() noexcept
+    {
+        sz = 0;
+        data_[0] = '\0';
+    }
+
+    //!\copydoc seqan3::small_vector::push_back(value_type const value)
+    constexpr void push_back(char const value) noexcept
+    {
+        assert(sz < capacity_);
+        data_[sz] = value;
+        ++sz;
+        data_[sz] = '\0';
+    }
+
+    //!\copydoc seqan3::small_vector::pop_back
+    constexpr void pop_back() noexcept
+    {
+        assert(sz > 0);
+        --sz;
+        data_[sz] = '\0';
+    }
+
+    //!\copydoc seqan3::small_vector::resize(size_type const)
+    constexpr void resize(size_type const count) noexcept
+    {
+        resize(count, '\0');
+    }
+
+    //!\copydoc seqan3::small_vector::resize(size_type const, value_type const value)
+    constexpr void resize(size_type const count, char const value) noexcept
+    {
+        assert(count <= capacity_);
+
+        for (size_t i = count; i <= sz; ++i) // sz >= count; discard values in [count, sz]
+            data_[i] = '\0';
+
+        for (size_t i = sz; i < count; ++i)  // sz < count; add `value` in [sz, count)
+            data_[i] = value;
+
+        sz = count;
+        data_[sz] = '\0';
+    }
+    //!\}
+
+    /*!\name Concatenation
+     * \{
+     */
+    /*!\brief Concatenates two small_strings by returning a new small_string.
+     * \param lhs The left-hand-side to concat with.
+     * \param rhs The right-hand-side to concat with.
+     * \returns `small_string<capacity_ + capacity2>` The new small_string with size `capacity_` + `capacity2`.
+     *
+     * ### Exceptions
+     *
+     * No-throw guarantee.
+     *
+     * ### Complexity
+     *
+     * Linear in the size of the strings.
+     */
+    template <size_t capacity2>
+    constexpr friend small_string<capacity_ + capacity2> operator+(small_string const & lhs,
+                                                                   small_string<capacity2> const & rhs) noexcept
+    {
+        small_string<capacity_ + capacity2> tmp{lhs};
+        tmp.insert(tmp.end(), rhs.begin(), rhs.end());
+        return tmp;
+    }
+    //!\}
+
+    /*!\name Implicit conversion to std::string
+     * \{
+     */
+    operator std::string() const
+    {
+        return str();
+    }
+    //!\}
```

Comment:
You documented this as a group, but not the member itself whihc will give a warning with newer doxygen. Also I would group it with `.str()` and `.c_str()`.